content/references/kata-trainer.md

```diff
@@ -3,41 +3,78 @@ title: Kata Trainer
+If you do not wish to complete the kata, you can skip to the next. A new kata will be chosen for you at random, based on your active [training routine](/getting-started/finding-kata#personal-trainer). You can always come back and continue where you left off, provided you have run the tests at least once.
 
 ### Unlock Solutions
-If you wish to give up trying to complete the kata on your own, you can unlock the solutions and view other warrior's fully completed solutions. If you choose to go this route, you will forfeit any honor and rank progression that you could have otherwise have earned for completing the kata.
+If you decide to give up your attempt at completing the kata, you can unlock and view other users' completed solutions. Choosing this option means you will **forfeit** any honor and rank progression you would have otherwise earned for completing the kata, even if you do so at a later date.
+
+### Discuss
+This button leads you to the kata's [discourse page](/concepts/kata/discourse), where you can view and post comments, questions and issues. Note that choosing this means leaving the kata trainer, causing any unsaved editor contents to be lost.
 
-### Shortcuts
+### Reset
+Revert the kata trainer to its original state as you found it at the beginning of your training. Choosing this action will reset the contents of both the solution and the sample tests editors to that provided by the kata author, _resulting in the loss of any code you have written_.
 
-When completing a kata, there are a number of shortcuts that you can use:
+### Test
+This button will run your solution against the contents of the sample tests editor, offering a quick way to verify basic functionality and correctness of your code. It also ensures the contents of the solution editor are saved, should you leave the trainer and return later.
+
+### Attempt
+The attempt button will run your solution against the kata's full test suite. These test cases are not shown in the editor, but you will see their results in the Output panel. Passing all tests is a pre-requisite to submitting your solution and completing the kata. If your attempt is successful, this button will turn green and change its label to "Submit". Clicking it again will then submit your solution, marking the kata as completed, and take you to the solutions page. Any changes made to your code between these steps will require a new attempt.
+
+## Shortcuts
+
+The following keyboard shortcuts are available inside the kata trainer:
 
 - <kbd>Command</kbd>+<kbd>Enter</kbd> Attempt
 - <kbd>Command</kbd>+<kbd>'</kbd> Run sample tests
 - <kbd>Command</kbd>+<kbd>U</kbd> Toggle focus between editors
-- <kbd>Command</kbd>+<kbd>I</kbd> Cycle through the left side sections
+- <kbd>Command</kbd>+<kbd>I</kbd> Cycle between Instructions and Output in the view panel
+- <kbd>Esc</kbd> Toggle full screen mode for the editor currently with focus (in VIM mode, toggles *command-line mode* on and off)
```

Comment:
```suggestion
- <kbd>Esc</kbd> Toggle full screen mode on the focused editor
```

I don't think we need to mention Vim here (or just mentioned "except Vim"). Also, it's called `Normal` mode, and doesn't toggle.